docs/apache-airflow/executor/kubernetes.rst

```diff
@@ -170,6 +164,29 @@ Additionally, the Kubernetes Executor enables specification of additional featur
 .. @enduml
 .. image:: ../img/k8s-happy-path.png
 
+Comparison with CeleryExecutor
+------------------------------
+
+In contrast to CeleryExecutor, KubernetesExecutor does not require additional components such as Redis and Flower, but does require access to Kubernetes cluster.
+
+With KubernetesExecutor, each task runs in its own pod. The pod is created when the task is queued, and terminates when the task completes.
+Historically, in some cases this presented a resource utilization advantage over CeleryExecutor, where you needed a fixed number of
+long-running celery worker pods, whether or not there were tasks to run.
+
+However, the official Apache Airflow helm chart can automatically scale celery workers down to zero based on the number of tasks in the queue,
+so when using the official chart, this is no longer an advantage.
+
+With Celery workers you will tend to have less task latency because the worker pod is already up and running when the task is queued.  On the
+other hand, because multiple tasks are running in the same pod, with Celery you may have to be more mindful about resource utilization
+in your task design, particularly memory consumption.
+
+An positive of KubernetesExecutor is if you have long-running tasks.  With KubernetesExecutor, if you do a deployment while a task is running,
+the task will keep running until it completes (or times out, etc).  But with CeleryExecutor, provided you have set a grace period, the
+task will only keep running up until the grace period has elapsed, at which time the task will be terminated.
+
+Finally, note that it doesn't have to be either-or.  With CeleryKubernetesExecutor you have the best of both worlds.  Tasks by default will
+go to Celery workers.  But if you want a task to run with KubernetesExecutor, you send it to the ``kubernetes`` queue and it will
```

Comment:
```suggestion
Finally, note that it doesn't have to be either-or as with CeleryKubernetesExecutor you have the best of both worlds. Tasks by default will
go to Celery workers, however if you want a task to run with KubernetesExecutor, you send it to the ``kubernetes`` queue and it will
```